Firestore/core/src/firebase/firestore/util/exception.h

```diff
@@ -32,19 +32,63 @@
 // For internal programming errors, including internal argument checking, use
 // HARD_ASSERT or HARD_FAIL().
 
+#include <stdexcept>
 #include <string>
 
 #include "Firestore/core/src/firebase/firestore/util/string_format.h"
+#include "absl/base/attributes.h"
 
 namespace firebase {
 namespace firestore {
 namespace util {
-namespace internal {
 
-[[noreturn]] void ThrowIllegalState(const std::string& message);
-[[noreturn]] void ThrowInvalidArgument(const std::string& message);
+#if ABSL_HAVE_EXCEPTIONS
+/**
+ * An exception thrown if Firestore encounters an internal error.
+ */
+class FirestoreError : public std::logic_error {
+ public:
+  using std::logic_error::logic_error;
+};
+#endif  // ABSL_HAVE_EXCEPTIONS
+
+/**
+ * An enumeration of logical exception types mapping to common user visible
+ * exceptions we might throw in response do some invalid action in an
+ * interaction with the Firestore API.
+ */
+enum class Exception {
```

Comment:
The name seems to imply it's a class, maybe something like `ExceptionType`?

Reply:
Renamed to `ExceptionType`.